src/patternfly/components/ModalBox/modal-box.scss

```diff
@@ -2,163 +2,93 @@
   --pf-c-modal-box--MaxHeight: calc(100vh - var(--pf-global--spacer--2xl)); // MaxHeight ensures that the modal will not go off the screen and instead the body will scroll
-  --pf-c-modal-box--MarginRight: var(--pf-global--spacer--md);
-  --pf-c-modal-box--MarginLeft: var(--pf-c-modal-box--MarginRight);
+  --pf-c-modal-box--MarginX: var(--pf-global--spacer--md);
 
-  // Box Header variables
-  --pf-c-modal-box__close--Top: var(--pf-global--spacer--sm);
-  --pf-c-modal-box__close--Right: var(--pf-global--spacer--md);
+  // Modal Box Title variables
+  --pf-c-modal-box--c-title--MarginBottom: var(--pf-global--spacer--md);
 
   // Box Header variables
-  --pf-c-modal-box__header--PaddingTop: var(--pf-global--spacer--xl);
-  --pf-c-modal-box__header--PaddingRight: var(--pf-global--spacer--xl);
-  --pf-c-modal-box__header--PaddingBottom: var(--pf-global--spacer--sm);
-  --pf-c-modal-box__header--PaddingLeft: var(--pf-global--spacer--xl);
-
-  // Box Header Title
-  --pf-c-modal-box__header-title--FontSize: var(--pf-global--FontSize--3xl);
-  --pf-c-modal-box__header-title--FontWeight: var(--pf-global--FontWeight--normal);
-  --pf-c-modal-box__header-title--LineHeight: var(--pf-global--LineHeight--sm);
-
-  // Box Body variables
-  --pf-c-modal-box__body--PaddingTop: var(--pf-global--spacer--sm);
-  --pf-c-modal-box__body--PaddingRight: var(--pf-global--spacer--xl);
-  --pf-c-modal-box__body--PaddingBottom: var(--pf-global--spacer--sm);
-  --pf-c-modal-box__body--PaddingLeft: var(--pf-global--spacer--xl);
-  --pf-c-modal-box__body--MarginTop: var(--pf-global--spacer--xl);
+  --pf-c-modal-box--c-button--MarginLeft: var(--pf-global--spacer--sm);
+  --pf-c-modal-box--c-button--Top: #{pf-size-prem(-2px)};
 
   // Box Footer variables
-  --pf-c-modal-box__footer--PaddingTop: var(--pf-global--spacer--xl);
-  --pf-c-modal-box__footer--PaddingRight: var(--pf-global--spacer--xl);
-  --pf-c-modal-box__footer--PaddingBottom: var(--pf-global--spacer--xl);
-  --pf-c-modal-box__footer--PaddingLeft: var(--pf-global--spacer--xl);
+  --pf-c-modal-box__footer--MarginTop: var(--pf-global--spacer--xl);
 
   // Box Footer Button variables
-  --pf-c-modal-box__footer--ButtonSpacer: var(--pf-global--spacer--md); // Button spacer is used to manipulate margin-left and/or margin-right values at various breakpoints, with a single value.
+  --pf-c-modal-box__footer--c-button--MarginRight: var(--pf-global--spacer--md); // Button spacer is used to manipulate margin-left and/or margin-right values at various breakpoints, with a single value.
+  --pf-c-modal-box__footer--c-button--sm--MarginRight: calc(var(--pf-c-modal-box__footer--c-button--MarginRight) / 2);
 
   // This component always needs to be light
   @extend %pf-t-light;
 
   position: relative;
   z-index: var(--pf-c-modal-box--ZIndex);
-  display: flex;
```

Comment:
Not sure if you meant to remove this? There are still `flex` properties on the children, and you want to ensure the body still scrolls. I'm also curious what `--pf-c-modal-box--MinHeight` is all about. It's quite the var and I'm not sure why it's necessary? I don't see anything about a `min-height` in the issue https://github.com/patternfly/patternfly-next/issues/210 or PR https://github.com/patternfly/patternfly-next/pull/301. Maybe @srambach can chime in?

Reply:
Assuming you keep `flex` on this, that's going to break the positioning of the close icon. If we can't figure the close button out soon, I'm inclined just to leave it's positioning the way it was (though we can still remove the `.pf-c-modal-box__close` element) and just make the CSS refactoring updates. We can always go back and create new issues to make sure the components are inline with the sketch file.

Reply:
My bad I didn't realize that taking off flex affected the scroll on the body. I updated the code so that the body could scroll without flex on the parent? It seems to work just the same. I also removed min-height because it wasn't affecting anything but I'll wait until @srambach confirms.

Reply:
Just as it says in the comment, MinHeight was there to make sure that the ModalBox was tall enough that content was not obscured past the bottom of the box. The use of it seems to have been removed at some point, so it's not actually getting used. However, a very short viewport does cause content to overrun the box, so I'd recommend having a look at how to prevent that.

Reply:
@srambach thanks, I gave the body section a `min-height` of at least 1 line of text. That seems to keep the body text from disappearing in a short viewport. The modal contents still overflow in a super short viewport. It isn't obvious how (to me anyways) how to keep the modal height as tall as the content in it. `min-height: min-content` doesn't allow the body section to shrink/overflow.  The problem with the old `min-height` is that it accounted for a header and footer being present, so if neither of those are present, the modal's min-height is too tall. I chatted with @mceledonia and he says it's enough of an edge case that we can follow up on after beta. What do you think?